legal-api/src/legal_api/reports/report.py

```diff
@@ -374,12 +375,29 @@ def _format_alteration_data(self, filing):
             filing['shareClasses'] = filing['alteration']['shareStructure'].get('shareClasses', [])
             filing['resolutions'] = filing['alteration']['shareStructure'].get('resolutionDates', [])
 
-        # Get previous business type
-        meta_data = self._filing.meta_data or {}
-        prev_legal_type = meta_data.get('alteration', {}).get('fromLegalType')
+        if self._filing.status == 'COMPLETED':
+            meta_data = self._filing.meta_data or {}
+            prev_legal_type = meta_data.get('alteration', {}).get('fromLegalType')
+            new_legal_type = meta_data.get('alteration', {}).get('toLegalType')
+            prev_legal_name = meta_data.get('alteration', {}).get('fromLegalName')
+            to_legal_name = meta_data.get('alteration', {}).get('toLegalName')
+        else:
+            prev_legal_type = filing.get('business').get('legalType')
+            new_legal_type = filing.get('alteration').get('business').get('legalType')
+            prev_legal_name = filing.get('business').get('legalName')
+            identifier = filing.get('business').get('identifier')
+            name_request_json = filing.get('alteration').get('nameRequest')
+            to_legal_name = name_request_json.get('legalName', identifier[2:] + ' B.C. LTD.')
```

Comment:
I believe line 390 will error out if there is no `nameRequest` block in the alteration filing.